src/pages/LoginPage.jsx

```diff
@@ -0,0 +1,151 @@
+import CircularProgress from "@mui/material/CircularProgress";
+import React, { useEffect, useRef, useState } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { login } from "../apis/user";
+import AlertBox from "../components/common/accounts/AlertBox";
+import InputGroup from "../components/common/accounts/InputGroup";
+import Box from "../components/common/atoms/Box";
+import Button from "../components/common/atoms/Button";
+import Container from "../components/common/atoms/Container";
+import useInput from "../hooks/useInput";
+import { fetchUserInfo, logIn } from "../store/slices/userSlice";
+
+export default function LoginPage() {
+  const navigate = useNavigate();
+  const [errorMessage, setErrorMessage] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false); // login api 호출 중인지 아닌지 확인
+  const { values, handleChange } = useInput({
+    email: "",
+    password: "",
+  });
+  const emailInputRef = useRef(null);
+  const passwordInputRef = useRef(null);
+  const dispatch = useDispatch();
+
+  const isValidValue = () => {
+    const emailRegex =
+      /([\w-.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/;
+    const passwordRegex =
+      /^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,20}$/;
+
+    return emailRegex.test(values.email) && passwordRegex.test(values.password);
+  };
+
+  const handleLogin = async () => {
+    if (!values.email) {
+      console.log(values.email);
+      setErrorMessage("이메일을 입력해주세요.");
+      emailInputRef.current.focus();
+      return;
+    }
+    if (!values.password) {
+      setErrorMessage("비밀번호를 입력해주세요.");
+      passwordInputRef.current.focus();
+      return;
+    }
+    if (!isValidValue()) {
+      // validation check fail시 api 호출하지 않음
+      setErrorMessage("");
+      setIsSubmitting(true);
+      await new Promise((resolve) => {
+        setTimeout(resolve, 500);
+      });
+      setErrorMessage("이메일 또는 비밀번호를 잘못 입력했습니다. ");
+      setIsSubmitting(false);
+      return;
+    }
+    try {
+      setIsSubmitting(true);
+      const response = await login({
+        // 여기서 localstorage에 token 저장
+        email: values.email,
+        password: values.password,
+      });
+      if (response.success) {
+        dispatch(logIn());
+        dispatch(fetchUserInfo());
+        navigate("/");
+      }
+      setIsSubmitting(false);
+    } catch (error) {
+      console.log(error);
+      setErrorMessage("이메일 또는 비밀번호를 잘못 입력했습니다. ");
+      setIsSubmitting(false);
+    }
+  };
+
+  // 첫 렌더링 시 email input에 focus
+  useEffect(() => {
+    if (emailInputRef.current) {
+      emailInputRef.current.focus();
+    }
```

Comment:
```suggestion
    emailInputRef.current?.focus();
```
optional chaining으로 축약 가능한 구문입니다!

Reply:
수정하였습니다!